lib/src/mutation.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:flutter/widgets.dart';
+
+class VoltMutation<T> {
+  final Future<bool> Function(T params) mutationFn;
+  final void Function(T? params)? onSuccess;
+  final void Function(T? params)? onError;
+  final ValueNotifier<bool> _isLoading;
+
+  VoltMutation({
+    required this.mutationFn,
+    this.onSuccess,
+    this.onError,
+    required ValueNotifier<bool> isLoading,
+  }) : _isLoading = isLoading;
+
+  Future<bool> mutate(T params) async {
```

Comment:
I think the mutation object (at least for RQ) also returned the state. This is useful when you're not using the callbacks to understand what the mutation state is. Maybe you have a useEffect instead. Otherwise the hook could just return the mutate function